### PR TITLE
core: autodetect more ls params

### PR DESCRIPTION
**Pull request by efriis**

### CODE REVIEW

File: libs/core/langchain_core/language_models/chat_models.py [hunk 1]
```diff
@@ -473,9 +473,37 @@ def _get_ls_params(
...
+            ls_params["ls_model_name"] = self.model_name
+
+        # temperature
+        if hasattr(self, "temperature") and isinstance(self.temperature, float):
+            ls_params["ls_temperature"] = self.temperature
+        elif "temperature" in kwargs and isinstance(kwargs["temperature"], float):
+            ls_params["ls_temperature"] = kwargs["temperature"]
+
+        # max_tokens
+        if hasattr(self, "max_tokens") and isinstance(self.max_tokens, int):
```
Comment (efriis) [inline libs/core/langchain_core/language_models/chat_models.py:502]: could also look for `max_tokens_to_sample` (community anthropic) but might be better to just override in that class

one other place to look for all of these is in `self.model_kwargs` if it exists

File: libs/core/langchain_core/language_models/chat_models.py [hunk 2]
```diff
@@ -473,9 +473,37 @@ def _get_ls_params(
         **kwargs: Any,
     ) -> LangSmithParams:
         """Get standard params for tracing."""
-        ls_params = LangSmithParams(ls_model_type="chat")
+
+        # get default provider from class name
+        default_provider = self.__class__.__name__
+        if default_provider.startswith("Chat"):
+            default_provider = default_provider[4:]
```
Comment (ccurme) [inline libs/core/langchain_core/language_models/chat_models.py:480]:
```suggestion
            default_provider = default_provider[4:].lower()
```

File: libs/core/langchain_core/language_models/chat_models.py [hunk 3]
```diff
@@ -473,9 +473,37 @@ def _get_ls_params(
...
+        # model
+        if hasattr(self, "model") and isinstance(self.model, str):
+            ls_params["ls_model_name"] = self.model
+        elif hasattr(self, "model_name") and isinstance(self.model_name, str):
+            ls_params["ls_model_name"] = self.model_name
+
+        # temperature
+        if hasattr(self, "temperature") and isinstance(self.temperature, float):
+            ls_params["ls_temperature"] = self.temperature
+        elif "temperature" in kwargs and isinstance(kwargs["temperature"], float):
```
Comment (ccurme) [inline libs/core/langchain_core/language_models/chat_models.py:498]: here and below: should we prefer parameter in kwargs over the (default) attribute?

Comment (efriis) [reply libs/core/langchain_core/language_models/chat_models.py:498]: smart

**Review by ccurme — APPROVED** [review]

**Pull request merged**